client/src/app/Pages/user-search/user-search.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { UserService } from '../../Services/user.service';
+
+@Component({
+  selector: 'app-user-search',
+  templateUrl: './user-search.component.html',
+  styleUrls: ['./user-search.component.css'],
+})
+export class UserSearchComponent implements OnInit {
+  searchForm: FormGroup;
+  users: any[] = [];
+  userRequestedFields: string[] = [];
+  searchResult: string | null = null;
+  allUserDtoFields: string[] = [
+    'Id',
+    'UserName',
+    'Email',
+    'CreatedAt',
+    'PhoneNumber',
+  ];
+  currentPage: number = 1;
+  pageSize: number = 10;
+  hasMoreData: boolean = true;
+
+  dropdownList: string[] = this.allUserDtoFields;
+  selectedItems: string[] = this.allUserDtoFields;
+  dropdownSettings = {};
+
+  constructor(private fb: FormBuilder, private userService: UserService) {
+    this.searchForm = this.fb.group({
+      sortBy: ['CreatedAt', [Validators.required]],
+      searchValue: [''],
+      ascending: [true],
+      fields: [this.selectedItems],
+    });
+  }
+
+  ngOnInit(): void {
+    this.dropdownSettings = {
+      singleSelection: false,
+      idField: 'item',
+      textField: 'item',
+      selectAllText: 'Select all',
+      unSelectAllText: 'Unselect all',
+      allowSearchFilter: true,
+    };
+    this.handleSearch();
+  }
+
+  handleSearch(page: number = 1): void {
```

Comment:
Same here

Reply:
I have craeted a separate method to construct a query